library/src/amd_detail/rocblaslt/src/tensile_host.cpp

```diff
@@ -1192,6 +1206,23 @@ rocblaslt_status runContractionProblem(rocblaslt_handle                   handle
         hardware     = Tensile::hip::GetDevice(*deviceProp);
 
         std::shared_ptr<TensileDataGemm> data = std::static_pointer_cast<TensileDataGemm>(gemmData);
+        int requestedAlgoCount = 1, returnAlgoCount;
+        rocblaslt_matmul_heuristic_result heuristicResultsArray[requestedAlgoCount];
+        if (algo == nullptr) {
+            status = getBestSolutions(prob,
+                                      handle,
+                                      gemmData,
+                                      requestedAlgoCount,
+                                      heuristicResultsArray,
+                                      &returnAlgoCount,
+                                      prob.workspaceSize);
+            if (returnAlgoCount == 0) return rocblaslt_status_not_implemented;
+            algo = &heuristicResultsArray[0].algo;
+            if (prob.workspaceSize == 0 || prob.workspace == nullptr) {
+                prob.workspaceSize = heuristicResultsArray[0].workspaceSize;
+                CHECK_HIP_ERROR(hipMalloc(&prob.workspace, prob.workspaceSize));
```

Comment:
don't allocate workspace in hipblaslt runtime.
getBestSolutions already has the workspace size info to get the proper solution